docs/proposals/container-runtime-interface/capabilities.md

```diff
@@ -0,0 +1,221 @@
+
+```
+message Capability {
+    // List of capabilities to retain.
+    repeated string requested_capabilities = 1;
+}
+```
+
+## Use Cases
+
+### User specifies `requestedSet` only in the pod spec:
+
+The simplest use case is when a user only sets the `requestedSet` field in a new pod spec:
+
+```
+  spec:
+    containers:
+      securityContext:
+        capabilities:
+          requestedSet:
+          - MKNOD, NET_ADMIN, SETPCAP, AUDIT_WRITE, ...
+```
+
+The resolved requested set equals `requestedSet`.
+
+### User specifies `add`/`drop` but no `requestedSet` in the pod spec:
+
+The second primary use case will be dealing with backwards-compatibility with existing pod specs that may specify `add` or `remove`:
+
+```
+  spec:
+    containers:
+      securityContext:
+        capabilities:
+          add:
+          - NET_ADMIN
+          drop:
+          - MKNOD
+```
+
+The resolved requested set is calculated in the API server based on a _default set_.
+
+The _default set_ is determined as follows:
+
+- The API server has a hard-coded set of capabilities (for illustrative purposes, called `HardcodedDefaultCapabilities`)
+- If admission control is not enabled, or admission control is enabled but there are no capabilities-related fields set on the `PodSecurityPolicySpec`, then `HardcodedDefaultCapabilities` is used as the default set.
+- Else (i.e. if admission control is enabled and the `PodSecurityPolicySpec` (PSP) has any capabilities-related fields set), then:
+   - If the `DefaultCapabilities` field is set, that is merged with any `DefaultAddCapabilities` and `RequiredDropCapabilities` set on the same `PodSecurityPolicySpec`.
+   - Else (i.e. the `DefaultCapabilities` field is not set), then:
+       - If the submitted pod's `SecurityContext` has a `requested-set` field, that is merged with any `DefaultAddCapabilities` and `RequiredDropCapabilities` set on the same `PodSecurityPolicySpec`.
+       - Else, `HardcodedDefaultCapabilities` is merged with any `DefaultAddCapabilities` and `RequiredDropCapabilities` set on the same `PodSecurityPolicySpec`.
+
+The _resolved requested set_ is then calculated by merging the _default set_ with the `add` and `drop` fields on the submitted pod's `SecurityContext` (i.e. "default set" + `add` - `drop`).
+
+Finally, the `requestedSet` field is set to "default set" - `add` - `drop` in the pod spec stored in the API server (which converts this pod spec into the form in the use case below).
+This ensures that all pod spec objects returned from the API server have a populated `requestedSet` field that preserves the immutability invariant.
+
+
+### User specifies both `requestedSet`, and `add`/`drop` in the pod spec:
+
+This use case is a bit contrived and primarily presented to demonstrate backwards-compatibility and the converted form from the previous use case:
```

Comment:
It's actually not that contrived. If the user starts with `add`/`drop`, the `requsetedSet` will be resolved & set. The pod now has `requestedSet` and `add`/`drop`. Further, if the user then edits the pod spec, the 3 might not be in agreement.